### PR TITLE
Finalize index page 

### PR DESCRIPTION
**Pull request by SamZhang02**
Index page is pretty much done. 

Changes:
- Made magnifying glass icon slightly smaller
- Added a parser to parse `&amp;` in course titles
- Added links to each search result suggestion in the format of `/cource/${result._id}`
- Removed the nav bar buttons from the template and added a button for About page
- Capped the search results' array size to 6 to avoid overflowing the screen 

Screenshot 2023-03-25 at 2 40 08 PM

### CODE REVIEW

File: client/src/pages/Home.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const Home = () => {
 
   const handleInputChange = async (query: string) => {
     try {
-      setResults(await fetchClient.getData<Course[]>(`/search?query=${query}`));
+      setResults((await fetchClient.getData<Course[]>(`/search?query=${query}`)).slice(0, 6));
```
Comment (terror) [inline client/src/pages/Home.tsx:18]: We don't need to slice courses here, we can limit the # of courses suggested on the backend, see https://github.com/terror/mcgill.gg/blob/master/crates/db/src/db.rs#L104

File: client/src/components/CourseSearchBar.tsx [hunk 2]
```diff
@@ -12,13 +12,14 @@ export const CourseSearchBar = ({
...
   const [searchSelected, setSearchSelected] = useState(false);
+  const parser = new DOMParser();
 
   return (
     <div className='relative'>
       <div className='relative w-full mt-4'>
         <div className='absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none'>
           <Search
-            size={20}
+            size={19}
```
Comment (terror) [inline client/src/components/CourseSearchBar.tsx:22]: I think size 20 is fine, i meant the literal size of the circle in the icon was large -- we can leave this for now

File: client/src/pages/Home.tsx (hunk 1 above)
Comment (SamZhang02) [reply client/src/pages/Home.tsx:18]: dope, ill make a change to that method then. 10 items seem to overflow any screen thats not my 32inch monitor.

Comment (terror) [reply client/src/pages/Home.tsx:18]: hmm, what about adding a scroll overflow to the search results div then? 10 seems like a good amount to show, so we should just let users scroll through them if it overflows

Comment (SamZhang02) [reply client/src/pages/Home.tsx]: Sure, it was already scroll overflow, so I just reversed the array slice

Comment (terror) [reply client/src/pages/Home.tsx]: actually i just tried this out and the overflow seems fine, users can still scroll the page to see all of the results

**Review by terror — APPROVED** [review]

**Pull request merged**